### PR TITLE
edpm_compute switch from centos to edpm-hardened-uefi

### PR DESCRIPTION
**Pull request by steveb**
This switches a crc environment to the edpm-hardened-uefi.qcow2 image
which is distributed in a container image (build defined in [1]).

Automation will rebuild the edpm-hardened-uefi image as required, so the
container image
quay.io/podified-antelope-centos9/edpm-hardened-uefi:current-podified
will remain up to date.

[1] https://github.com/openstack-k8s-operators/edpm-image-builder/

### CODE REVIEW

File: devsetup/scripts/gen-edpm-compute-node.sh [hunk 1]
```diff
@@ -157,17 +158,16 @@ cat <<EOF >${OUTPUT_BASEDIR}/${EDPM_COMPUTE_NAME}.xml
 EOF
 
 cat <<EOF >${OUTPUT_BASEDIR}/${EDPM_COMPUTE_NAME}-firstboot.sh
-growpart /dev/vda 1
-xfs_growfs /
+# extend GPT partition to the end of the disk, can be removed when this lands:
+# https://review.opendev.org/c/openstack/diskimage-builder/+/827558
+sgdisk -e /dev/vda
```
Comment (steveb) [inline devsetup/scripts/gen-edpm-compute-node.sh:163]: the diskimage-builder change has landed, so there would be no need for any firstboot.sh with this image now

File: devsetup/scripts/gen-edpm-compute-node.sh [hunk 2]
```diff
@@ -199,7 +195,8 @@ chmod +x ${OUTPUT_BASEDIR}/${EDPM_COMPUTE_NAME}-firstboot.sh
 if [ ! -f ${DISK_FILEPATH} ]; then
     if [ ! -f ${CRC_POOL}/${BASE_DISK_FILENAME} ]; then
         pushd ${CRC_POOL}
-        curl -L -k ${CENTOS_9_STREAM_URL} -o ${BASE_DISK_FILENAME}
+        podman run --volume ${CRC_POOL}:/target:rw --rm ${EDPM_IMAGE}
```
Comment (steveb) [inline devsetup/scripts/gen-edpm-compute-node.sh:198]: Podman doesn't have permission to write this file, I need to try this in a real environment to figure it out

**Comment by steveb** [comment]
I think this is ready to be reviewed on its merits now, so removing the WIP. install-yamls-deploy must pass before this lands, but any failures should be more down to the image itself

**Comment by steveb** [comment]
The permissions issue should be fixed now, but there is now an issue with getting the IP address assigned

**Comment by danpawlik** [comment]
recheck

**Comment by danpawlik** [comment]
recheck

**Comment by danpawlik** [comment]
recheck

File: devsetup/scripts/gen-edpm-compute-node.sh [hunk 3]
```diff
@@ -30,12 +30,12 @@ if [ ${STANDALONE} = "true" ]; then
...
 EDPM_COMPUTE_VCPUS=${EDPM_COMPUTE_VCPUS:-2}
 EDPM_COMPUTE_RAM=${EDPM_COMPUTE_RAM:-4}
-EDPM_COMPUTE_DISK_SIZE=${EDPM_COMPUTE_DISK_SIZE:-20}
+EDPM_COMPUTE_DISK_SIZE=${EDPM_COMPUTE_DISK_SIZE:-30}
 EDPM_COMPUTE_NETWORK=${EDPM_COMPUTE_NETWORK:-default}
 EDPM_COMPUTE_NETWORK_TYPE=${EDPM_COMPUTE_NETWORK_TYPE:-network}
 DATAPLANE_DNS_SERVER=${DATAPLANE_DNS_SERVER:-192.168.122.1}
-CENTOS_9_STREAM_URL=${CENTOS_9_STREAM_URL:-"https://cloud.centos.org/centos/9-stream/x86_64/images/CentOS-Stream-GenericCloud-9-latest.x86_64.qcow2"}
-BASE_DISK_FILENAME=${BASE_DISK_FILENAME:-"centos-9-stream-base.qcow2"}
+EDPM_IMAGE=${EDPM_IMAGE:-"quay.io/podified-antelope-centos9/edpm-hardened-uefi:current-podified"}
```
Comment (SeanMooney) [inline devsetup/scripts/gen-edpm-compute-node.sh:37]: would this not prevent use form testing with vanila centos locallly.

have you tested this change with the old image?

**Comment by steveb** [comment]
I proposed this before the baremetal job existed. It is not required now

**Pull request closed without merging**